apps/src/blockly/customBlocks/googleBlockly/proceduresBlocks.js

```diff
@@ -146,3 +150,115 @@ GoogleBlockly.Extensions.registerMutator(
+ * @param {WorkspaceSvg} workspace The workspace containing procedures.
+ * @returns an array of block objects representing the flyout blocks
+ */
+export function flyoutCategory(workspace) {
+  const blockList = [];
+
+  const newFunctionButton = {
+    kind: 'button',
+    text: msg.createBlocklyFunction(),
+    callbackKey: 'createNewFunction',
+  };
+
+  const functionDefinitionBlock = {
+    kind: 'block',
+    type: 'procedures_defnoreturn',
+    fields: {
+      NAME: Blockly.Msg.PROCEDURES_DEFNORETURN_PROCEDURE,
+    },
+  };
+
+  const createNewFunction = function () {
+    // Everything here is place-holder code that should be replaced with a
+    // call to open the behavior editor with a new defintion block.
+    // Until then, we just create a block under all existing blocks on the
+    // main workspace.
+    createAndCenterNewDefBlock(functionDefinitionBlock);
+  };
+
+  if (useModalFunctionEditor) {
+    workspace.registerButtonCallback('createNewFunction', createNewFunction);
+    blockList.push(newFunctionButton);
+  } else {
+    blockList.push(functionDefinitionBlock);
+  }
+
+  const allWorkspaces = Blockly.Workspace.getAll().filter(
+    workspace => !workspace.isFlyout
+  );
+  const allFunctionNames = [];
+  allWorkspaces.forEach(workspace => {
+    const behaviorBlocks = workspace
+      .getTopBlocks()
+      .filter(topBlock => topBlock.type === 'procedures_defnoreturn');
+    behaviorBlocks.forEach(block =>
+      allFunctionNames.push(block.getFieldValue('NAME'))
+    );
+  });
+  allFunctionNames.sort().forEach(name => {
+    blockList.push({
+      kind: 'block',
+      type: 'procedures_callnoreturn',
+      extraState: {
+        name: name,
+      },
+      fields: {
+        NAME: name,
+      },
+    });
+  });
+
+  return blockList;
+}
+
+const getLowestBlockBottomY = () => {
+  let lowestBlockBottom = 0;
+  Blockly.getMainWorkspace()
+    .getTopBlocks()
+    .forEach(block => {
+      const blockY = block.getRelativeToSurfaceXY().y;
+      const blockBottom = blockY + block.getHeightWidth().height;
+      if (blockBottom > lowestBlockBottom) {
+        lowestBlockBottom = blockBottom;
+      }
+    });
+  return lowestBlockBottom + 16;
+};
+
+export const createAndCenterNewDefBlock = function (blockState) {
+  // Everything here is place-holder code that should be replaced with a
+  // call to open the behavior editor with a new defintion block.
+  // Until then, we just create a block under all existing blocks on the
+  // main workspace.
+
+  const newDefBlock = Blockly.serialization.blocks.append(
+    {...blockState, x: 16, y: getLowestBlockBottomY()},
+    Blockly.getMainWorkspace()
+  );
+  // Hide the open toolbox flyout
+  Blockly.getMainWorkspace().hideChaff();
+  // Scroll to the new block and select it.
+  Blockly.getMainWorkspace().centerOnBlock(newDefBlock.id);
+  newDefBlock.select();
+};
+
+export function sortProceduresByName(a, b) {
```

Comment:
@molly-moen This function is never used. I assume Mike might've meant for it to be used [here](https://github.com/code-dot-org/code-dot-org/pull/53154/files#diff-8c6c333b74263db24d773ecbfd5cd4d66708e6cdd5ec730094f783cf9f501937R206) and [here](https://github.com/code-dot-org/code-dot-org/pull/53154/files#diff-18f81b893f4283e51973795b9e9f866e83c6a4017bb5e3251f16d2384c644be7R196). Thoughts?